### DISCUSSION TITLE
Action fails to login into Docker registry 'ghcr.io' - only works manually

### DISCUSSION THREAD

**Discussion by ghost** [community/community#30753] [Actions]
I co-maintain a private repository and we employ a self-hosted runner for our GitHub action. Since yesterday I receive failed CI messages on every re-run due to an access problem which I cannot understand.

```
##[debug]Starting: Set up job
Current runner version: '2.295.0'
Runner name: 'xxxxxxxxx-TestEnvironment'
Runner group name: 'Default'
Machine name: 'xxxxxxxxxxxxxxxxxx'
GITHUB_TOKEN Permissions
Secret source: Actions
##[debug]Primary repository: xxxxxxxxxxxxxxxxxxx
Prepare workflow directory
##[debug]Update context data
##[debug]Evaluating job-level environment variables
##[debug]Evaluating job container
##[debug]Evaluating: github.actor
##[debug]Evaluating Index:
##[debug]..Evaluating github:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'actor'
##[debug]=> 'hjokunbus'
##[debug]Result: 'hjokunbus'
##[debug]Evaluating: secrets.CONTAINER_READ
##[debug]Evaluating Index:
##[debug]..Evaluating secrets:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'CONTAINER_READ'
##[debug]=> '***'
##[debug]Result: '***'
##[debug]Evaluating job service containers
##[debug]Evaluating job defaults
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v3' (SHA:2541b1294d2704b0964813337f33b291d3f8596b)
##[debug]Download 'https://api.github.com/repos/actions/checkout/tarball/2541b1294d2704b0964813337f33b291d3f8596b' to '/home/holger/actions-runner/_work/_actions/_temp_ae5d4458-4a8a-4888-92f7-5ba3331451d3/2c4fa50f-a6d8-4e53-9eaa-b409fca112bd.tar.gz'
##[debug]Unwrap 'actions-checkout-2541b12' to '/home/holger/actions-runner/_work/_actions/actions/checkout/v3'
##[debug]Archive '/home/holger/actions-runner/_work/_actions/_temp_ae5d4458-4a8a-4888-92f7-5ba3331451d3/2c4fa50f-a6d8-4e53-9eaa-b409fca112bd.tar.gz' has been unzipped into '/home/holger/actions-runner/_work/_actions/actions/checkout/v3'.
##[debug]action.yml for action: '/home/holger/actions-runner/_work/_actions/actions/checkout/v3/action.yml'.
##[debug]Set step '__run' display name to: 'Clean the workspace'
##[debug]Set step '__actions_checkout' display name to: 'Run actions/checkout@v3'
##[debug]Set step '__run_2' display name to: 'Test with clang-tidy'
##[debug]Collect running processes for tracking orphan processes.
##[debug]Finishing: Set up job
0s

##[debug]Evaluating condition for step: 'Initialize containers'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Initialize containers
##[debug]Register post job cleanup for stopping/deleting containers.
Checking docker version
  /usr/bin/docker version --format '{{.Server.APIVersion}}'
  '1.39'
  Docker daemon API version: '1.39'
  /usr/bin/docker version --format '{{.Client.APIVersion}}'
  '1.39'
  Docker client API version: '1.39'
Clean up resources from previous jobs
  /usr/bin/docker ps --all --quiet --no-trunc --filter "label=82336b"
  /usr/bin/docker network prune --force --filter "label=82336b"
Create local container network
  /usr/bin/docker network create --label 82336b github_network_680e367e3ca74e168b62ad703860f04f
  c338b7456f973cc0cba2cff9a77691bb7d90bb85c7022adcf60a88592397fa7b
Starting job container
  /usr/bin/docker --config /home/holger/actions-runner/_work/_temp/.docker_e54ffea8-e56e-460e-867a-bc788bd55961 login ghcr.io -u hjokunbus --password-stdin
  Error: Docker login for 'ghcr.io' failed with exit code 1
  ##[debug]System.InvalidOperationException: Docker login for 'ghcr.io' failed with exit code 1
  ##[debug]   at GitHub.Runner.Worker.ContainerOperationProvider.ContainerRegistryLogin(IExecutionContext executionContext, ContainerInfo container)
  ##[debug]   at GitHub.Runner.Worker.ContainerOperationProvider.StartContainerAsync(IExecutionContext executionContext, ContainerInfo container)
  ##[debug]   at GitHub.Runner.Worker.ContainerOperationProvider.StartContainersAsync(IExecutionContext executionContext, Object data)
  ##[debug]   at GitHub.Runner.Worker.JobExtensionRunner.RunAsync()
  ##[debug]   at GitHub.Runner.Worker.StepsRunner.RunStepAsync(IStep step, CancellationToken jobCancellationToken)
  ##[debug]Finishing: Initialize containers
```

The secret has no expiring date and still should be valid - I can successfully login with docker manually using it. The action worked just fine the day before yesterday, we also did not run into any limits, since the pipeline runs only every few weeks. And the error message leaves me clueless about what the problem might be. The runner machine is accessible, the `actions-runner/runsvc.sh` script is in the list of active processes.

What can I check to figure out what the problem might be?

**Comment by dayananda30** [comment]
Hey @hjokunbus , we are facing the issue in our environment.
Did you find a way to solve this problem?

**Reply by ghost** [reply]
Unfortunately I have no clue how to solve this, since there is no change from our side related to it. I've added [an issue to the Github module most likely responsible](https://github.com/docker/login-action/issues/262). So far I haven't received any response. And as a temporary workaround we moved our pipeline to the cloud version, which unfortunately is more restricted by payment options.
I would be delighted to get any response by the Github people how to address this.

**Reply by daltonv** [reply]
Hey @hjokunbus I'm not sure about the exact error you are receiving, but it looks like you are using a custom PAT for reading from packages and adding that to your secrets. Is that right?

If so, maybe try explicitly adding the repo running the action as a repo that can use the action in the packages settings:
![image](https://user-images.githubusercontent.com/8504933/188897120-ce8e198b-50d7-4bb3-8c25-cf02fc56f1c6.png)

 Then your can just login with the auto generated `GITHUB_TOKEN` like shown below.

```yaml
- name: Login to GitHub Packages
  uses: docker/login-action@v1
  with:
        registry: ghcr.io
        username: ${{ github.actor }}
        password: ${{ secrets.GITHUB_TOKEN }}
```

**Reply by dayananda30** [reply]
we are facing this issue intermittently.

**Reply by ghost** [reply]
Yes, it is a custom personal token. And it worked for months. Would you mind elucidating why I need to change the repo's config in order to fix the problem? I didn't get the memo of PAT being revoked/unusable for actions.

**Reply by daltonv** [reply]
@hjokunbus I'm really not sure why you would be having an issue. I was just suggesting you use the [GITHUB_TOKEN](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) to see if another method would work. I find it easier to work with those tokens then private PATs anyway.

The repo config only needs to be altered for the auto generated GITHUB_TOKEN https://docs.github.com/en/packages/learn-github-packages/configuring-a-packages-access-control-and-visibility#ensuring-workflow-access-to-your-package

**Comment by sayhiben** [comment]
This was [a service issue](https://www.githubstatus.com/incidents/d181frs643d4) on GitHub's side that went unacknowledged for several hours. @github-staff, our production CI/CD systems rely on this product. Your customers need better post-mortems and guarantees of stability than we're seeing.

**Reply by ghost** [reply]
I have no idea what this is about. The issue still is occuring.

**Comment by kristof-mattei** [comment]
Same here. Facing the issue intermittently. Both with using `crane` and `buildx`.

**Comment by ghost** [comment]
Possible related issue at [GitHub actions](https://github.com/actions/runner/issues/1978).

**Comment by chandrakanthbala** [comment]
Hi, we are facing same issue with our self hosted runners. We have docker images in GHCR. We are seeing this issue intermittently (more often), but we are not able to reproduce this error on demand. Is there any resolution for this?

We are using runner version 2.292. Is this a runner software issue?
